src/wp-includes/theme.php

```diff
@@ -4084,6 +4084,5 @@ function create_initial_theme_features() {
  * @return boolean Whether the current theme is a block-based theme or not.
  */
 function wp_is_block_template_theme() {
-	return is_readable( get_theme_file_path( '/block-templates/index.html' ) ) ||
-		is_readable( get_theme_file_path( '/templates/index.html' ) );
```

Comment:
`is_block_based()` doesn't have this check for `/templates/index.html`, so not sure if we should use `wp_get_theme()→is_block_based()` as the implementation for `wp_is_block_template_theme()`. (I don't know what `/templates/index.html` is.)

Reply:
I've reverted this change for now. Let's look at simplifying in a follow-up after beta 1.

Reply:
@noisysocks 

>(I don't know what /templates/index.html is.)

I think that's because we use `/templates/` instead of `/block-templates` folder to store templates now.
More info can be found here: https://github.com/WordPress/wordpress-develop/commit/b90fc1b6dda3bfff668ad07e25da545c3eb10423
I've updated the `WP_Theme::is_block_based` method, so now it works the same way as the current implementation of `wp_is_block_template_theme`.
I think it's not critical for WordPress 5.9 whether we use `wp_get_theme()→is_block_based()` internally or the current implementation (checking files with `is_readable` function). `wp_is_block_template_theme`'s internal implementation can be anything we want. We will not break backward compatibility if we change that implementation later.
I'd like to open a new trac issue to get more feedback on changing the implementation of `wp_is_block_template_theme` and `get_theme_file_path` (because this PR introduces new `WP_Theme::get_file_path` method as well).
In my opinion this PR can be merged now.